src/Tools/ConfigurationSchemaGenerator/ConfigSchemaEmitter.cs

```diff
@@ -719,6 +715,32 @@ private static string[] CreateExclusionPaths(List<string>? exclusionPaths)
         return result;
     }
 
+    private static void ReplaceNodeWithKeyCasingChange(JsonObject jsonObject, string key, JsonNode value)
+    {
+#if NET9_0_OR_GREATER
```

Comment:
@eiriktsarpalis - is this the recommended way to respond to the breaking change? Do we have documentation guidance on the break?

I almost wonder if we should be checkign the System.Text.Json major version number instead. Or doing some sort of "runtime capability test" - for example, execute a method that checks for the new behavior and set a `bool` flag.

The situation that can break this is if you are targeting `net8.0`, but somehow the `9.0` System.Text.Json nuget package gets pulled into the app (ex. through a transitive reference). In that situation, the `#else` code will be compiled into the app, but the 9.0 STJ behavior will happen.

Reply:
I'm not sure I fully understand what tests were broken by the particular change in STJ 9, but per the discussion in https://github.com/dotnet/runtime/issues/108790#issuecomment-2407508284 the pre-9 behaviour is undocumented, accidental behaviour that is inconsistent with how all other dictionary types work in the same context. 

We've agreed to add a breaking change note in our documentation for completeness, but beyond that I don't think it's appropriate to introduce mitigations down the stack just so that the pre-9 behavior is preserved. Assuming there are failing tests that check against that behavior, then I would recommend just updating the tests.

Reply:
We've moved to always use STJ `9.0.0` even on `net8.0`. So we no longer have to switch behavior here.